PermissionsScrapper/PermissionsScrapper/PermsDescription.cs

```diff
@@ -0,0 +1,141 @@
+
+                var result = GetAuthentication(config);
+
+                if (result != null)
+                {
+                    var scopesDescriptionsJson = GetScopesDescriptionsJson(config, result);
+
+                    if (!scopesDescriptionsJson.Equals(_scopesDescriptionsJson, StringComparison.OrdinalIgnoreCase))
+                    {
+                        _scopesDescriptionsJson = scopesDescriptionsJson;
+
+                        // TODO: Call GitHub API with json file to upload to devx-content-repo
+                    }
+                }
+                else
+                {
+                    log.LogInformation($"Failed to get authentication at: {DateTime.Now}");
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogInformation($"Exception occurred: {ex.InnerException.Message ?? ex.Message}\r\n Time of occurrence: {DateTime.Now}");
+            }
+        }
+
+        /// <summary>
+        /// Gets authentication to a protected Web API.
+        /// </summary>
+        /// <param name="config">The application configuration settings.</param>
+        /// <returns>An authentiction result, if successful.</returns>
+        private static AuthenticationResult GetAuthentication(ApplicationConfig config)
+        {
+            IConfidentialClientApplication app;
+            app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
+                  .WithClientSecret(config.ClientSecret)
+                  .WithAuthority(new Uri(config.Authority))
+                  .Build();
+
+            string[] scopes = new string[] { $"{config.ApiUrl}.default" };
+
+            try
+            {
+                return app.AcquireTokenForClient(scopes)
+               .ExecuteAsync().GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        /// <summary>
+        /// Retrieves permissions and their descriptions from a Service Principal
+        /// </summary>
+        /// <param name="config">The application configuration settings.</param>
+        /// <param name="result">The JSON response of the permissions and their descriptions retrieved from the Service Prinicpal.</param>
+        /// <returns></returns>
+        private static string GetScopesDescriptionsJson(ApplicationConfig config, AuthenticationResult result)
+        {
+            try
+            {
+                var httpClient = new HttpClient();
+                var apiCaller = new ProtectedApiCallHelper(httpClient);
+                var spJson = apiCaller
+                    .CallWebApiAsync($"{config.ApiUrl}{config.ApiVersions[0]}/serviceprincipals?$filter=appId eq '{config.ServicePrincipalId}'", result.AccessToken)
+                    .GetAwaiter().GetResult(); // fetch for v1.0 ; no business case for fetching for beta yet
+
+                var cleanedSpJson = CleanJsonData(spJson, config);
```

Comment:
this can be null from the call to CallWebApiAsync so you should probably check

Reply:
Resolved